pymeasure/instruments/princetoninstruments/princetonSP2300i.py

```diff
@@ -0,0 +1,697 @@
+        match = re.match(
+            fr'^\s*({re.escape(self._last_command)})?\s*(.*?)\s*(ok)?(\?)?\s*$',
+            reply
+        )
+
+        # Check read is OK
+        if match[3] == 'ok':
+            self._read_ok = True
+        else:
+            self._read_ok = False
+
+        # Reset default timeout
+        self.adapter.connection.timeout = self._timeout_default
+
+        return match[2]
+
+    def wait_for(self, query_delay=0):
+        """Wait for some time.
+
+        :param query_delay: override the global query_delay.
+        """
+        super().wait_for(query_delay or self.query_delay)
+
+    def write_bytes(self):
+        """Not implemented -- use adapter method."""
+        raise NotImplementedError("Use adapter method")
+
+    def read_bytes(self):
+        """Not implemented -- use adapter method."""
+        raise NotImplementedError("Use adapter method")
+
+    def write_binary_values(self):
+        """Not implemented -- use adapter method."""
+        raise NotImplementedError("Use adapter method")
+
+    def read_binary_values(self):
+        """Not implemented -- use adapter method."""
+        raise NotImplementedError("Use adapter method")
+
+    def check_errors(self):
+        """Read all errors from the instrument and log them.
+
+        :return: List of error entries.
+        """
+        errors = []
+        if not self._read_ok:
+            # IO error
+            err = [0, "Communication error. Reset device and flush buffer. "]
+            log.error(f"{self.name}: {err[0]}, {err[1]}")
+            errors.append(err)
+        return errors
+
+    def check_get_errors(self):
+        """Check for errors after having gotten a property and log them.
+
+        Called if :code:`check_get_errors=True` is set for that property.
+
+        :return: List of error entries.
+        """
+        return self.check_errors()
+
+    def check_set_errors(self):
+        """Check for errors after having set a property and log them.
+
+        Called if :code:`check_set_errors=True` is set for that property.
+
+        :return: List of error entries.
+        """
+        self.read()
+        return self.check_errors()
+
+    ################
+    #  Properties  #
+    ################
+
+    # ?NM
+    wavelength = Instrument.measurement(
+        "?NM", "Measure the current wavelength position [nm].",
+        get_process=lambda v: ureg.Quantity(v),  # convert to quantity
+    )
+
+    # MONO-?DONE
+    is_scanning = Instrument.measurement(
+        "MONO-?DONE", "Get the scanning status.",
+        map_values=True,
+        values={True: 0, False: 1},
+    )
+
+    # NM/MIN
+    # ?NM/MIN
+    scan_rate = Instrument.control(
+        "?NM/MIN", "%0.3f NM/MIN",
+        """Control scan rate in wavelength/time.
+
+            Set as a pint compatible value.
+        """,
+        get_process=lambda v: ureg.Quantity(v),  # convert to quantity
+        set_process=lambda v: ureg.Quantity(v).m_as('nm/min'),
```

Comment:
For ease of use, could you add some method, which assumes the correct unit, for example something like:
~~~python
def generate_magnitude_extractor(units)
    def get_quantity_magnitude(value):
        if isinstance(value, ureg.Quantity):
            return value.m_as(units)
        else:
            return value
    return get_quantity_magnitude


scan_rate = Instrument.control(...
    set_process=generate_magnitude_extractor("nm/min")
~~~

That will make it easier to use for people who do not use pint (yet).

For another instrument with pint, see #929